### DISCUSSION TITLE
Terrible filter/search

### DISCUSSION THREAD

**Discussion by lacerated** [layday/instawow#81] [Ideas]
Guess any1 who tried to find addons with instawow knows: its impossible to find anything unless u have exact name of the addon.
Any plans to fix it?

**Comment by layday** [comment]
There are improvements that can be made to the fuzzy matcher but it'll never be as good as full-text search.  You don't need to have the exact name of the add-on, but a naïve edit distance-based implementation will suffer when the search query is significantly shorter or longer than the add-on name.  I intend to replace the algorithm I have in place with weighted ratio from [RapidFuzz](https://github.com/maxbachmann/RapidFuzz).

**Reply by lacerated** [reply]
Let me give you simple example. I was trying to find addon for Midsummer event. Searching with keyword "Midsummer" returned 23 addons with no word Midsummer in name or description, but if i search "HandyNotes Midsummer Festival" it will be in top result.
Usually i just go to curse, wowinterface or google and search for exact name of the addon and then use instawow to install it.

**Reply by layday** [reply]
Indeed, that is the problem I described - there's a 20 character difference between "Midsummer" and "HandyNotes Midsummer Festival".  With the new matcher, searching for "Midsummer" returns the HandyNotes add-on as the first result:

<img width="651" alt="Screenshot 2021-06-22 at 17 49 17" src="https://user-images.githubusercontent.com/31134424/122946600-2c74e680-d382-11eb-9c78-c7823a4aec11.png">

**Reply by lacerated** [reply]
wow looks awesome, thanks m8!

**Reply by layday** [reply]
Released as part of v1.26.0.
